synapse/handlers/pagination.py

```diff
@@ -72,13 +73,123 @@ def __init__(self, hs):
         self.auth = hs.get_auth()
         self.store = hs.get_datastore()
         self.clock = hs.get_clock()
+        self.config = hs.config
```

Comment:
As a general rule, we prefer to reference things *within* the config rather than add a reference to the whole Config object, where possible. It means a better chance of finding errors during load time rather than hours later.

It looks like the only thing you need here is `retention_default_max_lifetime`, so I think there are a couple of options. You could have `self._retention_default_max_lifetime = hs.config.retention_default_max_lifetime` here; alternatively you could make `retention_default_max_lifetime` another argument to `purge_history_for_rooms_in_range`